tests/test_client_functional.py

```diff
@@ -987,6 +987,60 @@ async def redirect(request):
     resp.close()
 
 
+async def test_HTTP_307_REDIRECT_POST_FORMDATA(aiohttp_client: Any) -> None:
+    async def handler(request):
+        return web.Response(text=request.method)
+
+    async def redirect(request):
+        await request.read()
+        raise web.HTTPTemporaryRedirect(location="/")
+
+    app = web.Application()
+    app.router.add_post("/", handler)
+    app.router.add_post("/redirect", redirect)
+    client = await aiohttp_client(app)
+
+    data = FormData()
+    data.add_field(
+        'file',
+        io.BytesIO(),
+        filename='bytes_file',
+    )
+    resp = await client.post("/redirect", data=data)
+    assert 200 == resp.status
+    assert 1 == len(resp.history)
+    txt = await resp.text()
+    assert txt == "POST"
+    resp.close()
+
+
+async def test_HTTP_308_PERMANENT_REDIRECT_POST_FORMDATA(aiohttp_client: Any) -> None:
+    async def handler(request):
+        return web.Response(text=request.method)
+
+    async def redirect(request):
+        await request.read()
+        raise web.HTTPPermanentRedirect(location="/")
+
+    app = web.Application()
+    app.router.add_post("/", handler)
+    app.router.add_post("/redirect", redirect)
+    client = await aiohttp_client(app)
+
+    data = FormData()
+    data.add_field(
+        'file',
+        io.BytesIO(),
+        filename='bytes_file',
+    )
+    resp = await client.post("/redirect", data=data)
+    assert 200 == resp.status
+    assert 1 == len(resp.history)
+    txt = await resp.text()
+    assert txt == "POST"
+    resp.close()
```

Comment:
Please merge these tests into one with `@pytest.mark.parametrize()`

Reply:
Okay, I'll add a few more to describe how I see the desired behavior.